sdk/internal/mock/mock.go

```diff
@@ -58,6 +60,27 @@ func NewTLSServer() (*Server, func()) {
 	return s, func() { s.srv.Close() }
 }
 
+// NewTLSHTTP2Server creates a new Server object with HTTP2 configuration.
+// The returned close func must be called when the server is no longer needed.
+// It will return nil for both the server and close func if it encountered an error
+// when configuring HTTP2 for the new TLS server.
+func NewTLSHTTP2Server() (*Server, func()) {
+	s := newServer()
+	s.srv = httptest.NewUnstartedServer(http.HandlerFunc(s.serveHTTP))
+	if err := http2.ConfigureServer(s.srv.Config, new(http2.Server)); err != nil {
+		return nil, nil
+	}
+	s.srv.TLS = s.srv.Config.TLSConfig
+	s.srv.StartTLS()
+	return s, func() { s.srv.Close() }
+}
+
+// ServerConfig returns the server config. Please note this should not be
+// modified since Start or StartTLS has already been called on the server.
+func (s *Server) ServerConfig() *http.Server {
```

Comment:
I assume this is exposed so tests can access it.  Curious which part(s) needed to fix the issue you were working on?

Reply:
Yes, the test looks something like this
```
srv, close := mock.NewTLSHTTP2Server()
	defer close()
	tr := &http.Transport{TLSClientConfig: srv.ServerConfig().TLSConfig}
	if err := http2.ConfigureTransport(tr); err != nil {
		t.Fatalf("Failed to configure http2 transport: %v", err)
	}
	tr.TLSClientConfig.InsecureSkipVerify = true
	client := &http.Client{Transport: tr}
	srv.AppendResponse(mock.WithBody([]byte(accessTokenRespSuccess)))
	options := DefaultInteractiveBrowserCredentialOptions()
	options.Options = &TokenCredentialOptions{AuthorityHost: srv.URL(), HTTPClient: client}
```

Reply:
I'm following the "proper" way to configure a new server, according to this: https://github.com/golang/go/issues/34939